proto/forwarding/forwarding_table.proto

```diff
@@ -60,66 +61,64 @@ enum TableType {
 // the previous extension).
 message TableDesc {
   // Required type of the table.
-  optional TableType table_type = 1;
+  TableType table_type = 1;
 
-  // Optional set of default actions. If no actions are specified, the table
+  //  set of default actions. If no actions are specified, the table
   // marks the packet to continue packet processing.
   repeated ActionDesc actions = 2;
 
   // Required id of the table.
-  optional TableId table_id = 4;
+  TableId table_id = 3;
 
-  // Required extension to the table.
-  extensions 100 to max;
+  oneof table {
+    ExactTableDesc exact = 4;
+    PrefixTableDesc prefix = 5;
+    FlowTableDesc flow = 6;
+    BridgeTableDesc bridge = 7;
+  }
 }
 
 // An EntryDesc describes how a table entry is identified.
 // Note that an extension must be always used. Each type of entry is an
 // extension with a unique extension number (1 greater than
 // the previous extension).
```

Comment:
These comments are still remaining, but feel free to mark a TODO and clean up later.

Reply:
fixed